godtossing/Pos/Ssc/GodTossing/Seed.hs

```diff
@@ -6,38 +8,40 @@ module Pos.Ssc.GodTossing.Seed
-import           Pos.Ssc.GodTossing.Core      (Commitment (..),
-                                               CommitmentsMap (getCommitmentsMap),
-                                               OpeningsMap, SharesMap, getOpening,
-                                               secretToSharedSeed, verifyOpening)
-import           Pos.Ssc.GodTossing.Error     (SeedError (..))
-import           Pos.Ssc.GodTossing.Functions (vssThreshold)
-import           Pos.Util.Util                (getKeys)
-
+import           Control.Lens             (_Left)
+import qualified Data.HashMap.Strict      as HM
+import qualified Data.HashSet             as HS
+
+import           Pos.Binary.Class         (AsBinary, fromBinaryM)
+import           Pos.Core                 (SharedSeed, StakeholderId, addressHash, mkCoin,
+                                           sumCoins, unsafeIntegerToCoin)
+import           Pos.Crypto               (DecShare, Secret, VssPublicKey, recoverSecret,
+                                           verifySecret)
+import           Pos.Lrc.Types            (RichmenStakes)
+import           Pos.Ssc.GodTossing.Core  (Commitment (..), CommitmentsMap (..),
+                                           Opening (..), OpeningsMap, SharesMap,
+                                           SignedCommitment, getCommShares,
+                                           secretToSharedSeed, verifyOpening,
+                                           vssThreshold)
+import           Pos.Ssc.GodTossing.Error (SeedError (..))
+import           Pos.Util.Util            (getKeys)
 
 -- | Calculate SharedSeed. SharedSeed is a random bytestring that all
 -- nodes generate together and agree on.
---
--- TODO: do we need to check secrets' lengths? Probably not.
 calculateSeed
-    :: CommitmentsMap        -- ^ All participating nodes
-    -> OpeningsMap           -- ^ Openings sent by those nodes
-    -> SharesMap             -- ^ Decrypted shares
-    -> RichmenStakes          -- ^ How much stake nodes have
+    :: CommitmentsMap                         -- ^ All participating nodes
+    -> HashMap StakeholderId (AsBinary VssPublicKey) -- ^ Nodes' VSS keys
+    -> OpeningsMap                            -- ^ Openings sent by the nodes
+    -> SharesMap                              -- ^ Decrypted shares
+    -> RichmenStakes                          -- ^ How much stake nodes have
     -> Either SeedError SharedSeed
-calculateSeed commitments' openings lShares richmen = do
-    let commitments = getCommitmentsMap commitments' -- just unwrapping
-    let participants = getKeys commitments
+calculateSeed commitments' binVssKeys openings lShares richmen = do
+    let commitments :: HashMap StakeholderId SignedCommitment
+        commitments = getCommitmentsMap commitments' -- just unwrapping
+    let participants :: HashSet StakeholderId
+        participants = getKeys commitments
+    -- vssKeys :: HashMap StakeholderId VssPublicKey
```

Comment:
I think it would be better not to have this type commented out and instead have it on the next line.

Reply:
Okay, done.